src/Features/CSharp/Portable/ExtractMethod/CSharpMethodExtractor.CSharpCodeGenerator.cs

```diff
@@ -394,6 +399,71 @@ yield return
                 }
             }
 
+            /// <summary>
+            /// If the statement has an `out var` declaration expression for a variable which
+            /// needs to be removed, we need to turn it into a plain `out` parameter, so that
+            /// it doesn't declare a duplicate variable.
+            /// If the statement has a pattern declaration (such as `3 is int i`) for a variable
+            /// which needs to be removed, we will annotate it as a conflict, since we don't have
+            /// a better refactoring.
+            /// </summary>
+            private StatementSyntax FixDeclarationExpressionsAndDeclarationPatterns(StatementSyntax statement,
+                HashSet<SyntaxAnnotation> variablesToRemove)
+            {
+                var replacements = new Dictionary<SyntaxNode, SyntaxNode>();
+
+                var declarations = statement.DescendantNodes()
+                    .Where(n => n.IsKind(SyntaxKind.DeclarationExpression, SyntaxKind.DeclarationPattern));
+
+                foreach (var node in declarations)
+                {
+                    switch (node.Kind())
+                    {
+                        case SyntaxKind.DeclarationExpression:
+                            var declaration = (DeclarationExpressionSyntax)node;
+                            if (declaration.VariableComponent.Kind() != SyntaxKind.TypedVariableComponent)
+                            {
+                                break;
+                            }
+
+                            var variableComponent = (TypedVariableComponentSyntax)declaration.VariableComponent;
+                            if (variableComponent.Designation.Kind() != SyntaxKind.SingleVariableDesignation)
+                            {
+                                break;
+                            }
+
+                            var designation = (SingleVariableDesignationSyntax)variableComponent.Designation;
+                            var name = designation.Identifier.ValueText;
+                            if (variablesToRemove.HasSyntaxAnnotation(designation))
+                            {
+                                replacements.Add(declaration, SyntaxFactory.IdentifierName(designation.Identifier)
+                                    .WithLeadingTrivia(variableComponent.Type.GetTrailingTrivia()));
```

Comment:
I confirmed that `With...Trivia` indeed replace trivia (rather than append).
Latest commit fixes the issue although hard to test.